ext_localconf.php

```diff
@@ -3,27 +3,66 @@
     die();
 }
 
-(function () {
-    \TYPO3\CMS\Core\Utility\ExtensionManagementUtility::addPageTSConfig(
-        '<INCLUDE_TYPOSCRIPT: source="FILE:EXT:wv_deepltranslate/Configuration/TsConfig/Page/pagetsconfig.tsconfig">'
-    );
+(static function (): void {
+    if (!isset($GLOBALS['TYPO3_CONF_VARS']['SYS']['features']['deepltranslate.automaticTranslation'])) {
+        $GLOBALS['TYPO3_CONF_VARS']['SYS']['features']['deepltranslate.automaticTranslation'] = false;
+    }
 
-    $GLOBALS['TYPO3_CONF_VARS']['SC_OPTIONS']['Backend\Template\Components\ButtonBar']['getButtonsHook']['wv_deepltranslate'] =
-        \WebVision\WvDeepltranslate\Hooks\ButtonBarHook::class . '->getButtons';
     //allowLanguageSynchronizationHook manipulates l10n_state
     $GLOBALS['TYPO3_CONF_VARS']['SC_OPTIONS']['t3lib/class.t3lib_tcemain.php']['processDatamapClass'][]
         = \WebVision\WvDeepltranslate\Hooks\AllowLanguageSynchronizationHook::class;
 
-    //hook for translate content
-    $GLOBALS['TYPO3_CONF_VARS']['SC_OPTIONS']['t3lib/class.t3lib_tcemain.php']['processTranslateToClass']['deepl']
-        = \WebVision\WvDeepltranslate\Hooks\TranslateHook::class;
     //hook to checkModifyAccessList for editors
     $GLOBALS['TYPO3_CONF_VARS']['SC_OPTIONS']['t3lib/class.t3lib_tcemain.php']['checkModifyAccessList']['deepl']
         = \WebVision\WvDeepltranslate\Hooks\TCEmainHook::class;
 
+    // preview Hook
     $GLOBALS['TYPO3_CONF_VARS']['SC_OPTIONS']['tslib/class.tslib_fe.php']['contentPostProc-all']['deepl-1675946132'] =
         \WebVision\WvDeepltranslate\Hooks\DeeplPreviewFlagGeneratePageHook::class . '->renderDeeplPreviewFlag';
 
+    // register Icons
```

Comment:
I would suggest to do the icon registry within the `Configuration/Icons.php` file to do this as simple configuration. 

See: https://docs.typo3.org/m/typo3/reference-coreapi/main/en-us/ExtensionArchitecture/FileStructure/Configuration/Icons.html

As this extension currently still supports older version than v11, I would suggest to use a version check `if ((new Typo3Version())->getMajorVersion() < 11) {...}`, fetching that file and doing the registration based on it as as fall back with a `@deprecated Remove this after minimum core version support raised to v11`

Reply:
For icon registry feel free to a new pull request, as I only took the code above, as I want to handle, even if feature flag is set.